File: SDKs/amazonaws.com/shield/2016-06-02/python/src/sdk/models/shared/listattacksresponse.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from . import attacksummary


@dataclass_json
@dataclass
class ListAttacksResponse:
    attack_summaries: Optional[List[attacksummary.AttackSummary]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'AttackSummaries' }})
    next_token: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'NextToken' }})
    

File: SDKs/amazonaws.com/shield/2016-06-02/typescript/src/sdk/models/shared/listattacksresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { AttackSummary } from "./attacksummary";



export class ListAttacksResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=AttackSummaries", elemType: AttackSummary })
  attackSummaries?: AttackSummary[];

  @SpeakeasyMetadata({ data: "json, name=NextToken" })
  nextToken?: string;
}
